### PR TITLE
Slider to the map page.

### PR DESCRIPTION
**Pull request by realARCHaim**
https://youtrack.raccoongang.com/issue/OLGA-16

### CODE REVIEW

File: web/olga/analytics/models.py [hunk 1]
```diff
@@ -78,6 +78,15 @@ class InstallationStatistics(models.Model):
...
     unspecified_country_name = 'Country is not specified'
 
+    @staticmethod
+    def get_statistics_top_country(tabular_countries_list):
+        """
+        Get first country from tabular format country list.
+
+        List is sorted, first country is a top active students rank country.
+        """
+        return tabular_countries_list[0][0]
```
Comment (idegtiarov) [inline web/olga/analytics/models.py]: What happens if `tabular_countries_list == []`?

File: web/olga/analytics/models.py [hunk 2]
```diff
@@ -241,12 +296,18 @@ def get_students_per_country(cls):
         """
         Gather convenient and necessary data formats to render it from view.
         """
-        students_per_country = cls.get_students_per_country_stats()
+        months = cls.get_students_per_country_stats()
 
-        datamap_format_countries_list, tabular_format_countries_list = \
-            cls.create_students_per_country(students_per_country)
+        for _, month in months.iteritems():
```
Comment (idegtiarov) [inline web/olga/analytics/models.py]: Please omit `_` with `.values()` dict method

File: web/olga/analytics/models.py [hunk 3]
```diff
@@ -268,8 +329,8 @@ def get_students_countries_amount(cls):
 
         Actually `Country is not specified` field is not a country, so it does not fill up in countries amount.
         """
-        _, tabular_format_countries_list = cls.get_students_per_country()
-        countries_amount = len(tabular_format_countries_list) - 1
+        for month in months:
+            countries_amount = len(month['tabular_countries_list']) - 1
```
Comment (idegtiarov) [inline web/olga/analytics/models.py]: What about the case when `Country is not specified` was not received?

File: web/olga/charts/static/charts/slider.js [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+        return false;
+    });
+
+    prepare_months_table();
+    select_month(months_keys_sorted[months_keys_sorted.length - 1]);
+});
+
+function make_tr_from_tabular_countries(month_key, countries) {
+    countries_trs = '';
```
Comment (idegtiarov) [inline web/olga/charts/static/charts/slider.js]: Does `countries_trs` variable?

File: web/olga/charts/static/charts/slider.js [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+    for (var country_key in countries) {
+        country = countries[country_key];
```
Comment (idegtiarov) [inline web/olga/charts/static/charts/slider.js]: ditto

File: web/olga/charts/static/charts/slider.js [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    });
+
+    prepare_months_table();
+    select_month(months_keys_sorted[months_keys_sorted.length - 1]);
+});
+
+function make_tr_from_tabular_countries(month_key, countries) {
+    countries_trs = '';
+
+    for (var country_key in countries) {
```
Comment (idegtiarov) [inline web/olga/charts/static/charts/slider.js]: block scope local variable please use `let`

File: web/olga/charts/static/charts/slider.js [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+                <td class="text-right sorting_1" role="gridcell">${country[1][0]}</td>
+            </tr>
+        `;
+    }
+
+    return `<tbody id="tbody_${month_key}" style="display: none">${countries_trs}</tbody>`;
+}
+
+function select_month(month_key) {
+    var month = months[month_key]
```
Comment (idegtiarov) [inline web/olga/charts/static/charts/slider.js]: trailing semicolon
second question from what scope do you take variable months?

File: web/olga/analytics/models.py [hunk 8]
```diff
@@ -162,25 +171,71 @@ def overall_counts(cls):
...
+        )
 
-        world_students_per_country = defaultdict(int)
+        return cls.aggregate_countries_by_months(result_rows)
 
-        for instance_students in students_per_country:
-            for country, count in instance_students.iteritems():
-                world_students_per_country[country] += count
+    @classmethod
+    def aggregate_countries_by_months(cls, values_list):
```
Comment (idegtiarov) [inline web/olga/analytics/models.py]: Please add a description of the required parameter.

File: web/olga/analytics/models.py [hunk 9]
```diff
@@ -162,25 +171,71 @@ def overall_counts(cls):
...
+
+        for month_ordering, month_verbose, countries in values_list:
+            cls.add_month_countries_data(
+                month_ordering, month_verbose, countries, months
+            )
+
+        return months
+
+    @classmethod
+    def add_month_countries_data(
```
Comment (idegtiarov) [inline web/olga/analytics/models.py]: ditto

**Review by idegtiarov — CHANGES_REQUESTED** [review]

File: web/olga/analytics/models.py (hunk 3 above)
Comment (realARCHaim) [reply web/olga/analytics/models.py]: That's right. That bug was in the previous version. I can fix it in the current branch.

Comment (realARCHaim) [reply web/olga/analytics/models.py]: Found a bug here that the countries_amount doesn't iterates in the loop.

File: web/olga/charts/static/charts/slider.js (hunk 4 above)
Comment (realARCHaim) [reply web/olga/charts/static/charts/slider.js]: Yes, it is. Added an explicit variable definition into the function body.

File: web/olga/charts/static/charts/slider.js (hunk 7 above)
Comment (realARCHaim) [reply web/olga/charts/static/charts/slider.js]: months is in the global window scope

File: web/olga/analytics/models.py (hunk 9 above)
Comment (realARCHaim) [reply web/olga/analytics/models.py]: done

**Review by idegtiarov — DISMISSED** [review]

**Pull request merged**